cmd/gcp-controller-manager/app/node_annotater.go

```diff
@@ -207,3 +248,40 @@ func parseNodeURL(nodeURL string) (project, zone, instance string, err error) {
 	instance = parts[2]
 	return
 }
+
+// TODO: move this to instance.Labels. This is gross.
+func extractKubeLabels(instance *compute.Instance) (map[string]string, error) {
+	const labelsKey = "kube-labels"
+
+	if instance.Metadata == nil {
+		return nil, fmt.Errorf("instance %q has no labels")
+	}
+
+	var kubeLabels *string
+	for _, item := range instance.Metadata.Items {
+		if item == nil || item.Key != labelsKey {
+			continue
+		}
+		if item.Value == nil {
+			return nil, fmt.Errorf("instance %q had nil %q", instance.SelfLink, labelsKey)
```

Comment:
Same here - is it worth failing this annotator or maybe just glog.Errorf?

Reply:
The annotator can't proceed unless this value is non-nil. Failures are just glog.Errorf'd in the body of the annotator.